spock-tests/src/test/groovy/com/rackspacecloud/blueflood/repose/ReposeSpec.groovy

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2024 Rackspace US, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.repose
+
+import groovy.json.JsonBuilder
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.entity.StringEntity
+import org.apache.http.impl.client.HttpClients
+import spock.lang.IgnoreIf
+import spock.lang.Shared
+import spock.lang.Specification
+
+/*
+* Repose upgrade activity needs to validate various Repose filters features
+* This test can be run against local vagrant or stage or prod blueflood endpoints
+*
+* Vagrant (local) allows us to configure repose filters to tweak as needed.
+* Example: Rate limits, authorization needs
+*
+* Pre-requisites for running:
+* Environment variables:
+*  - TENANT_ID
+*  - X-Auth-Token - Rackspace identity token to work with keystone-v2 filter config
+*  - RATE_LIMIT_COUNT (Optional) - to test against vagrant or staging with configurable rate limit filters
+*  - ENV_TO_RUN (local/stage/prod) defaults to stage
+*/
+class ReposeSpec extends Specification {
+
+    final String X_AUTH_TOKEN_HEADER = "X-Auth-Token"
+
+    @Shared
+    String tenantId = System.getenv("TENANT_ID")
+    @Shared
+    String authToken = System.getenv("AUTH_TOKEN")
+    String envToRun = System.getenv("ENV_TO_RUN")?.toLowerCase() ?: "stage"
+    static def defaultRateLimit = Integer.parseInt(System.getenv("RATE_LIMIT_COUNT") ?: "0")
+
+    def httpClient = HttpClients.createDefault()
+    String ingestEndpointUrl
+    String queryEndpointUrl
+
+    def setupSpec() {
+        assert tenantId != null: envError("TENANT_ID")
+        assert authToken != null: envError("AUTH_TOKEN")
+    }
+
+    def envError(String variableName) {
+        "${variableName} environment variable is missing"
+    }
+
+    def setup() {
+        if (envToRun == "local") {
+            ingestEndpointUrl = "http://localhost:9001/v2.0/${tenantId}"
+            queryEndpointUrl = "http://localhost:9011/v2.0/${tenantId}"
+        } else {
+            def prefix = envToRun == "stage" ? "staging" : "global"
+            ingestEndpointUrl = "https://${prefix}.metrics-ingest.api.rackspacecloud.com/v2.0/${tenantId}"
+            queryEndpointUrl = "https://${prefix}.metrics.api.rackspacecloud.com/v2.0/${tenantId}"
+        }
+    }
```

Comment:
Isn't all of this spec setup, not per-test setup?

Reply:
Yes...done